packages/react-instantsearch/src/components/Menu.js

```diff
@@ -26,14 +29,17 @@ class Menu extends Component {
 
   renderItem = item => {
     const {refine, createURL} = this.props;
+    const label = item._highlightResult
```

Comment:
Second read: this is not an issue of always using Highlight but an issue of switching to isSearch ? `<Highlight..` : item.label;